crates/nu-command/tests/main.rs

```diff
@@ -23,3 +23,55 @@ fn quickcheck_parse(data: String) -> bool {
     }
     true
 }
+
+#[test]
+fn signature_name_matches_command_name() {
+    let ctx = crate::create_default_context();
+    let decls = ctx.get_decl_ids_sorted(true);
+    let mut failures = Vec::new();
+
+    for decl_id in decls {
+        let cmd = ctx.get_decl(decl_id);
+        let cmd_name = cmd.name();
+        let sig_name = cmd.signature().name;
+        let category = cmd.signature().category;
+
+        if cmd_name != sig_name {
+            failures.push(format!(
+                "{cmd_name} ({category:?}): Signature name \"{sig_name}\" is not equal to the command name \"{cmd_name}\""
+            ));
+        }
+    }
+
+    assert!(
+        failures.is_empty(),
+        "Name mismatch:\n{}",
+        failures.join("\n")
+    );
+}
```

Comment:
Can someone with deeper knowledge about the development history of the `Command` and `Signature` API crosscheck my assumptoins here? I don't want to have a test that might not be necessary.